cmake/modules/SIMDExt.cmake

```diff
@@ -31,18 +31,22 @@ if(CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64")
   if(HAVE_ARMV8_CRC)
     message(STATUS " aarch64 crc extensions supported")
   endif()
+  CHECK_C_COMPILER_FLAG(-march=armv8-a+crc+crypto HAVE_ARMV8_CRC_CRYPTO_INTRINSICS)
+  if(HAVE_ARMV8_CRC_CRYPTO_INTRINSICS)
+    set(ARMV8_CRC_COMPILE_FLAGS "${ARMV8_CRC_COMPILE_FLAGS} -march=armv8-a+crc+crypto")
```

Comment:
i have refined the patch by adding inline assembly so that the optimization will work even on lower GCC version which doesn't support crc&crypto intrinsics.  but if on higher GCC version supporting to the intrinsics, it will prefer intrinsics so that ARM compiler has a chance to further optimize it (such as better instruction schedule via more accurate pipeline latency calculation or adopting newest NEON instructions for better performance). besides performance, intrinsics will provide better readability. so the whole logic is now as follows:
```
if compiler supporting crc+crypto intrinsics then
    {optimization based on compiler intrinsics}
else if assembler supporting both crc and crypto instructions then
    {optimization based on crc+crypto instructions}
else if assembler only supporting crc instruction then
    {original optimization based linear crc instructions}
else
    {pure software implementation}
```